### PR TITLE
docs: improve windows.md

### PR DESCRIPTION
**Pull request by chriscool**
This adds a lot of steps that were missing and makes it possible to build on Windows.

License: MIT
Signed-off-by: Christian Couder <chriscool@tuxfamily.org>

### CODE REVIEW

File: docs/windows.md [hunk 1]
```diff
@@ -1,5 +1,50 @@
...
+The following commands should download or update go-ipfs and its
+dependencies:
+
+```sh
+go get -u github.com/whyrusleeping/gx
+go get -u github.com/whyrusleeping/gx-go
+go get -u github.com/ipfs/go-ipfs
+cd %GOPATH%/src/github.com/ipfs/go-ipfs
+gx --verbose install --global
+```
```
Comment (jbenet) [inline docs/windows.md]: should we put this in the makefile or something? it's annoying to ask people to have to install something. 

(or is it that Make has problems on windows?)

**Comment by jbenet** [comment]
LGTM one comment above

File: docs/windows.md (hunk 1 above)
Comment (chriscool) [inline docs/windows.md]: The problem is that installing "make" on Windows is usually done by installing Cygwin which takes a lot of time. We should perhaps check if there is a way to install make without Cygwin but it might be for another PR.

**Comment by whyrusleeping** [comment]
cc @RichardLitt

**Comment by chriscool** [comment]
There is the following error:

```
gx --verbose install --global
installing package: go-ipfs-0.4.0
  - QmZ4Qi3GaRbjcx28Sme5eMH7RQjGkt8wHxt2a65oLaeFEV not found locally, fetching into /home/travis/gopath/src/gx/ipfs/QmZ4Qi3GaRbjcx28Sme5eMH7RQjGkt8wHxt2a65oLaeFEV
  - fetching QmZ4Qi3GaRbjcx28Sme5eMH7RQjGkt8wHxt2a65oLaeFEV via ipfs api
  - QmZy2y8t9zQH2a1b8q2ZSLKp17ATuJoCNxxyMFG5qFExpt not found locally, fetching into /home/travis/gopath/src/gx/ipfs/QmZy2y8t9zQH2a1b8q2ZSLKp17ATuJoCNxxyMFG5qFExpt
  - fetching QmZy2y8t9zQH2a1b8q2ZSLKp17ATuJoCNxxyMFG5qFExpt via ipfs api
  - Qmazh5oNUVsDZTs2g59rq8aYQqwpss8tcUWQzor5sCCEuH not found locally, fetching into /home/travis/gopath/src/gx/ipfs/Qmazh5oNUVsDZTs2g59rq8aYQqwpss8tcUWQzor5sCCEuH
  - fetching Qmazh5oNUVsDZTs2g59rq8aYQqwpss8tcUWQzor5sCCEuH via ipfs api
  - QmQopLATEYMNg7dVqZRNDfeE2S1yKy8zrRh5xnYiuqeZBn not found locally, fetching into /home/travis/gopath/src/gx/ipfs/QmQopLATEYMNg7dVqZRNDfeE2S1yKy8zrRh5xnYiuqeZBn
  - fetching QmQopLATEYMNg7dVqZRNDfeE2S1yKy8zrRh5xnYiuqeZBn via ipfs api
  - QmYvsG72GsfLgUeSojXArjnU6L4Wmwk7wuAxtNLuyXcc1T not found locally, fetching into /home/travis/gopath/src/gx/ipfs/QmYvsG72GsfLgUeSojXArjnU6L4Wmwk7wuAxtNLuyXcc1T
  - fetching QmYvsG72GsfLgUeSojXArjnU6L4Wmwk7wuAxtNLuyXcc1T via ipfs api
  - QmUBogf4nUefBjmYjn6jfsfPJRkmDGSeMhNj4usRKq69f4 not found locally, fetching into /home/travis/gopath/src/gx/ipfs/QmUBogf4nUefBjmYjn6jfsfPJRkmDGSeMhNj4usRKq69f4
  - fetching QmUBogf4nUefBjmYjn6jfsfPJRkmDGSeMhNj4usRKq69f4 via ipfs api
ERROR: [1 / 6 ] parallel fetch: failed to fetch package: QmUBogf4nUefBjmYjn6jfsfPJRkmDGSeMhNj4usRKq69f4:get: 
ERROR: [2 / 6 ] parallel fetch: failed to fetch package: QmQopLATEYMNg7dVqZRNDfeE2S1yKy8zrRh5xnYiuqeZBn:get: 
ERROR: [3 / 6 ] parallel fetch: failed to fetch package: Qmazh5oNUVsDZTs2g59rq8aYQqwpss8tcUWQzor5sCCEuH:get: 
ERROR: [4 / 6 ] parallel fetch: failed to fetch package: QmYvsG72GsfLgUeSojXArjnU6L4Wmwk7wuAxtNLuyXcc1T:get: 
ERROR: [5 / 6 ] parallel fetch: failed to fetch package: QmZy2y8t9zQH2a1b8q2ZSLKp17ATuJoCNxxyMFG5qFExpt:get: 
ERROR: [6 / 6 ] parallel fetch: failed to fetch package: QmZ4Qi3GaRbjcx28Sme5eMH7RQjGkt8wHxt2a65oLaeFEV:get: 
ERROR: install deps: failed to fetch dependencies
make: *** [deps] Error 1
```

**Comment by chriscool** [comment]
Circle Ci has the following error in t0065-active-requests.sh:

```
expecting success: 
	grep "log/tail" cmd_out2 | grep "true" > /dev/null

not ok 10 - output looks good
#	
#		grep "log/tail" cmd_out2 | grep "true" > /dev/null
#	
```

**Comment by RichardLitt** [comment]
LGTM

**Comment by slothbag** [comment]
Thanks for this, I was able to get windows builds working again with minimal fuss. :+1:

**Comment by chriscool** [comment]
@slothbag great! Happy that it helped!
